### ISSUE TITLE
[BUG] Azure.Identity has a dependency to Windows.Forms (due to older MSAL) for all Windows builds

### ISSUE THREAD

**Issue by pedrolupin** [Azure/azure-sdk-for-net#44232]
### Library name and version

Azure.Identity

### Describe the bug

This is related to this MSAL bug which has been fixed:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4468

MSAL fixed it and released a new version, 4.61.0 a couple of weeks ago.

It seems that it's not yet being picked up by Azure.Identity 1.12 beta 3 yet.
Could you include MSAL 4.61.0 in your next release please?

### Expected behavior

Reference to MSAL 4.61.0 to avoid Desktop SDK in server installs.

### Actual behavior

Server build requires desktop SDK to run.

### Reproduction Steps

Azure.Identity 1.9.0 -> Microsoft.Identity.Client 4.49.1 did not have a dependency on System.Windows.Forms, but when updating to newer Azure.Identity 1.10.4 -> Microsoft.Identity.Client 4.56.0 the dependency to Windows.Forms appeared, this is a problem.

Azure.Identity should be able to run headless, on servers and many situations that don't have access to System.Windows.Forms because the .NET desktop runtime is not present.

**Comment by Ramo-Y** [comment]
We also had the same issue after updating `Azure.Identity`. The pipelines for the integration tests no longer ran because the .NET Desktop Runtime is not installed on the agents. The application did not run either, as the runtime is not available on the web server either.

We solved the problem by manually installing `Microsoft.Identity.Client` with version `4.61.1`, the `WindowsForms` references were then gone.

It took us time to find and fix the error, the dev environment was down during this time. It would have been helpful if this breaking change was described in the changelogs, I read through them before I made the updates.
